File: models/Student.js
const mongoose = require('mongoose');

const studentSchema = new mongoose.Schema({
  name: { type: String, required: true, trim: true },
  phone: { type: String, default: '' },
  institutionId: { type: mongoose.Schema.Types.ObjectId, ref: 'Institution', required: true, index: true },
  createdAt: { type: Date, default: Date.now },
});

studentSchema.index({ institutionId: 1, name: 1 });
studentSchema.index({ institutionId: 1, createdAt: -1 });

module.exports = mongoose.model('Student', studentSchema);

File: flutter_app/lib/models/student.dart
class Student {
  final String id;
  final String name;
  final String phone;
  final String? createdAt;

  Student({
    required this.id,
    required this.name,
    this.phone = '',
    this.createdAt,
  });

  factory Student.fromJson(Map<String, dynamic> json) {
    return Student(
      id: json['_id'] ?? '',
      name: json['name'] ?? '',
      phone: json['phone'] ?? '',
      createdAt: json['createdAt'],
    );
  }

  Map<String, dynamic> toJson() => {'name': name, 'phone': phone};
}
